src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -57,11 +59,82 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
                 }
             }
 
+            var controllerType = actionDescriptor.ControllerTypeInfo.AsType();
+            var propertyMetadata = new List<ModelMetadata>();
+            foreach (var property in actionDescriptor.CommonParameters)
+            {
+                var metadata = _modelMetadataProvider.GetMetadataForProperty(
+                    controllerType,
+                    propertyName: property.Name);
```

Comment:
One the next changes I wanted to make was to push things like `BindingSource, BinderName` into the MBC. So the idea is that `GetModelMetadataForParameter` would disappear, and instead of that, we would do a `GetMetadataForType` and use the descriptor + metadata for the type to set up the binding context.

The idea would be to keep it DRY by adding a constructor or factory method to MBC that does the needful based on MM. You'd only need to set properties if you want to override.